HW0/P6/P6.txt

```diff
@@ -1 +1,86 @@
-# Your answers here
+Problem 6.
+
+Count to Ten
+
+We are using 4 parallel processors to run the individual "Job"s. We would expect that the gap between starting and stopping a process would be 0.25 s, and the order that jobs start would be the order that jobs end. Furthermore, we would expect jobs to start and end sequentially as processors become available. 
+
+Expected output:
+Assuming no latency...
+Hi Job 0
+Hi Job 1
+Hi Job 2
+Hi Job 3 ### <0.25 s pause>
+Bye Job 0
+Hi Job 4
+Bye Job 1
+Hi Job 5
+Bye Job 2
+Hi Job 6
+Bye Job 3
+Hi Job 7 ### <0.25 s pause>
+Bye Job 4
+Hi Job 8
+Bye Job 5
+Hi Job 9
+Bye Job 6
+Bye Job 7 ### <0.25 s pause>
+Bye Job 8
+Bye Job 9
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+What we get in practice is not the expected ordering, although we obtain our "sorted" ouput in the end. This may be a consequence of differences in latency, compute time, and bandwidth between processors. 
```

Comment:
quote from student's user,patrickday9, "What happens in practice is that the program actually wakes new processes before closing the previous ones. This is worrisome when programming in parallel as we must be concerned about data dependancies."